pkg/reposync/conditions.go

```diff
@@ -109,10 +111,63 @@ func SetSyncing(rs *v1beta1.RepoSync, status bool, reason, message, commit strin
 	setCondition(rs, v1beta1.RepoSyncSyncing, conditionStatus, reason, message, commit, errorSources, errorSummary, lastUpdate)
 }
 
+// SetReconcilerFinalizing sets the ReconcilerFinalizing condition to True.
+func SetReconcilerFinalizing(rs *v1beta1.RepoSync, reason, message string) (updated bool) {
+	updated, _ = setCondition(rs, v1beta1.RepoSyncReconcilerFinalizing, metav1.ConditionTrue, reason, message, "", nil, singleErrorSummary, now())
+	return updated
+}
+
+// RemoveReconcilerFinalizing removes the ReconcilerFinalizing condition.
+func RemoveReconcilerFinalizing(rs *v1beta1.RepoSync) (updated bool) {
+	return removeCondition(rs, v1beta1.RepoSyncReconcilerFinalizing)
+}
+
+// SetReconcilerFinalizerFailure sets the ReconcilerFinalizerFailure condition.
+// If there are errors, the status is True, otherwise False.
+func SetReconcilerFinalizerFailure(rs *v1beta1.RepoSync, errs status.MultiError) (updated bool) {
+	var conditionStatus metav1.ConditionStatus
+	var reason, message string
+	var summary *v1beta1.ErrorSummary
+	if errs != nil {
+		conditionStatus = metav1.ConditionTrue
+		reason = "DestroyFailure"
+		message = errs.Error()
+		summary = summarizeFinalizerErrors(errs)
+	} else {
+		conditionStatus = metav1.ConditionFalse
+		reason = "DestroySuccess"
+		message = "Successfully deleted managed resource objects"
+		summary = nil
+	}
+	updated, _ = setCondition(rs, v1beta1.RepoSyncReconcilerFinalizerFailure,
+		conditionStatus, reason, message, "", nil, summary, now())
+	return updated
+}
+
+// RemoveReconcilerFinalizerFailure removes the ReconcilerFinalizerFailure condition.
+func RemoveReconcilerFinalizerFailure(rs *v1beta1.RepoSync) (updated bool) {
+	return removeCondition(rs, v1beta1.RepoSyncReconcilerFinalizerFailure)
+}
+
+// summarizeErrors summarizes the errors from `sourceStatus` and `syncStatus`,
+// and returns an ErrorSource slice and an ErrorSummary.
+// TODO: handle error truncation (requires new status.finalizer fields)
+func summarizeFinalizerErrors(errs status.MultiError) *v1beta1.ErrorSummary {
+	if errs == nil {
+		return nil
+	}
+	// TODO: handle error truncation (requires new status.finalizer fields)
```

Comment:
I don't actually know. The applier truncates its errors because a full list is available in `status.sync` and `status.source`. But I'm not sure if there's a max length for the condition message string. I figured It'd be ok to add later if we discovered we needed it. And I didn't want to add a new status field without discussing it first.

Reply:
Talked to Nan about this and it seems truncation was added to avoid the RSync from reaching max etcd object size when there are lots of errors from lots of objects.

This PR punts on that optimization, since we don't need to for the e2e tests. We can add it later.